sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -36,214 +47,217 @@ template <typename T> std::vector<uint8_t> ToUInt8Vector(T const& in)
 
   return outVec;
 }
+
+template <typename> struct UniqueHandleHelper;
```

Comment:
Do we need such helper types when using the RAII pattern with `BasicUniqueHandle`?

Reply:
You are right, they can be simplified now (I was using more verbose form for debugging). Plus I found out that nullptr check prior to free is not necessary - unique_ptr does the check.